backend/vouch/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.db.models import ObjectDoesNotExist
+from core.models.user import User
+
+# Create your models here.
+class vouch(models.Model):
+
+    vouching = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="",
+        related_name="vouchinguser",
+    )
+    vouchedfor = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The contributor",
+        related_name="vouchedforuser",
+    )
+    is_public = models.BooleanField(
+        default=False, null=False, help_text="Should the vouching be public?"
```

Comment:
By default, model fields are not null.

```suggestion
        default=False, help_text="Should the vouching be public?"
```

See: https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.Field.null